AutomatedTesting/Gem/PythonTests/largeworlds/landscape_canvas/EditorScripts/ComponentUpdates_UpdateGraph.py

```diff
@@ -90,113 +90,83 @@ def ComponentUpdates_UpdateGraph():
     import os
 
     import azlmbr.bus as bus
-    import azlmbr.editor as editor
     import azlmbr.entity as entity
     import azlmbr.legacy.general as general
     import azlmbr.landscapecanvas as landscapecanvas
     import azlmbr.math as math
     import azlmbr.prefab as prefab
 
     import editor_python_test_tools.hydra_editor_utils as hydra
+    from editor_python_test_tools.editor_entity_utils import EditorEntity
     from editor_python_test_tools.utils import Report
-    from editor_python_test_tools.utils import TestHelper as helper
 
     # Open a simple level and instantiate BushFlowerBlender.prefab
     hydra.open_base_level()
     transform = math.Transform_CreateIdentity()
     position = math.Vector3(64.0, 64.0, 32.0)
     transform.invoke('SetPosition', position)
     test_prefab_path = os.path.join("Assets", "Prefabs", "BushFlowerBlender.prefab")
-    prefab_result = prefab.PrefabPublicRequestBus(bus.Broadcast, 'InstantiatePrefab', test_prefab_path,
+    prefab_result = prefab.PrefabPublicRequestBus(bus.Broadcast, "InstantiatePrefab", test_prefab_path,
                                                   entity.EntityId(), position)
     Report.critical_result(Tests.prefab_instantiated, prefab_result.IsSuccess())
 
     # Find root entity in the loaded level
-    search_filter = entity.SearchFilter()
-    search_filter.names = ["LandscapeCanvas"]
-
-    # Allow a few seconds for matching entity to be found
-    helper.wait_for_condition(lambda: len(entity.SearchBus(bus.Broadcast, 'SearchEntities', search_filter)) > 0, 5.0)
-    lc_matching_entities = entity.SearchBus(bus.Broadcast, 'SearchEntities', search_filter)
-    prefab_root_id = lc_matching_entities[0]         #Entity with Landscape Canvas component
-    Report.critical_result(Tests.lc_entity_found, prefab_root_id.IsValid())
+    prefab_root_entity = EditorEntity.find_editor_entity("LandscapeCanvas")
+    Report.critical_result(Tests.lc_entity_found, prefab_root_entity.id.IsValid())
```

Comment:
Is prefab_root_entity.id safe to dereference if the entity isn't found?

Reply:
If the entity isn't found, the find_editor_entity method would assert and interrupt the test. I did update this to use the must_be_unique argument since we're only expecting 1 of each entity the test searches for.

Reply:
find_editor_entity will fail an assert if zero entities match the search so the pytest execution will terminate at that point with the assertion stack.
if you want to handle the not found then you would use .find_editor_entities which returns a list of matches or empty list if none. you'd then check the return list for empty with a simple if prefab_root_entities:
this seems safe since execution would terminate on line 114 if not found

Reply:
I was seeing assertions occur without any actual python execution haltage.   To be more specific, I would see asserts in C++ appear, but not propogate down to python.  Python would continue running.  This could be a different bug.